src/Data/Yaml.php

```diff
@@ -42,16 +44,15 @@ public static function decode($string): array
 			throw new InvalidArgumentException('Invalid YAML data; please pass a string');
 		}
 
-		// remove BOM
-		$string = str_replace("\xEF\xBB\xBF", '', $string);
-		$result = Spyc::YAMLLoadString($string);
+		return match (static::handler()) {
+			'symfony' => YamlSymfony::decode($string),
+			default   => YamlSpyc::decode($string)
+		};
+	}
 
-		if (is_array($result) === true) {
-			return $result;
-		}
+	public static function handler(): string
+	{
+		return App::instance(null, true)?->option('yaml') ?? 'spyc';
```

Comment:
```suggestion
		return App::instance(null, true)?->option('yaml.handler') ?? 'spyc';
```

Reply:
Good idea IMO. For now we don't have any other options in this namespace, but in the future we might have (like in the `markdown` namespace). If we already use a sub-option, we don't need to deal with naming conflicts here down the road.